File: app/src/main/java/com/chooloo/www/callmanager/ui/page/PagePresenter.java
package com.chooloo.www.callmanager.ui.page;

import androidx.recyclerview.widget.RecyclerView;

import com.chooloo.www.callmanager.ui.base.BasePresenter;


public class PagePresenter<V extends PageMvpView> extends BasePresenter<V> implements PageMvpPresenter<V> {

    @Override
    public void onDialNumberChanged(String number) {
        mvpView.loadNumber(number);
    }

    @Override
    public void onSearchTextChanged(String text) {
        mvpView.loadSearchText(text);
    }

    @Override
    public void onScrollStateChanged(int newState) {
        if (newState == (RecyclerView.SCROLL_STATE_DRAGGING | RecyclerView.SCROLL_STATE_SETTLING)) {
            mvpView.setSearchBarFocused(false);
        }
    }
}

File: app/src/main/java/com/chooloo/www/koler/ui/page/PagePresenter.kt
package com.chooloo.www.koler.ui.page

import androidx.recyclerview.widget.RecyclerView
import com.chooloo.www.koler.ui.base.BasePresenter

class PagePresenter<V : PageMvpView> : BasePresenter<V>(), PageMvpPresenter<V> {
    override fun onDialNumberChanged(number: String) {
        // TODO "implement"
    }

    override fun onSearchTextChanged(text: String) {
        // TODO implement
    }

    override fun onScrollStateChanged(state: Int) {
        if (state == RecyclerView.SCROLL_STATE_DRAGGING or RecyclerView.SCROLL_STATE_SETTLING) {
            mvpView?.setSearchBarFocused(false)
        }
    }
}